### PR TITLE
Removing unnecessary dependencies in DotnetDispatcher

### PR DESCRIPTION
**Pull request by ajaybhargavb**
Follow-up to https://github.com/aspnet/RazorTooling/pull/107

@NTaylorMullen

### CODE REVIEW

File: src/Microsoft.Extensions.DotnetToolDispatcher.Sources/project.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
   "shared": "*.cs",
   "dependencies": {
     "Microsoft.DotNet.Cli.Utils": "1.0.0-*",
-    "Microsoft.Extensions.PlatformAbstractions": "1.1.0-*",
+    "Microsoft.Extensions.PlatformAbstractions": "1.0.0-*",
```
Comment (pranavkm) [inline src/Microsoft.Extensions.DotnetToolDispatcher.Sources/project.json]: We should remove these dependencies entirely. Dependencies listed in shared src projects don't get added to the consuming project.

**Comment by pranavkm** [comment]
⌚️

**Comment by ajaybhargavb** [comment]
🆙 📅

File: src/Microsoft.Extensions.DotnetToolDispatcher.Sources/project.json (hunk 1 above)
Comment (NTaylorMullen) [inline src/Microsoft.Extensions.DotnetToolDispatcher.Sources/project.json]: Talked offline, this should stay.

**Comment by ajaybhargavb** [comment]
🆙 📅

**Comment by NTaylorMullen** [comment]
:shipit:

**Comment by pranavkm** [comment]
:shipit:

**Pull request merged**